habitat/plan.sh

```diff
@@ -0,0 +1,208 @@
+#######################################################
+# !!!              IMPORTANT REMINDER             !!! #
+#######################################################
+# Any changes to plan.sh related to installed gems    #
+# (eg 'without' flags, additions/removals) must       #
+# also be updated in omnibus/config/software/gems.rb  #
+#######################################################
+
+do_install() {
+  export ruby_bin_dir
+  ruby_bin_dir="$pkg_prefix/ruby-bin"
+
+  build_line "Creating bin directories"
+  mkdir -p "$ruby_bin_dir"
+
+  # TODO @afiune we need to add the 'chef' top-level command
+  ( cd "${SRC_PATH}/components/gems" || exit_with "unable to enter components/gems directory" 1
+    appbundle "chef-cli"  "changelog,docs,debug"
+    wrap_ruby_bin "chef-cli"
+
+    appbundle "chef-bin" "docgen,chefstyle"
+    wrap_ruby_bin "chef-client"
+    wrap_ruby_bin "chef-solo"
+    wrap_ruby_bin "chef-resource-inspector"
+    wrap_ruby_bin "chef-shell"
+
+    appbundle "chef" "docgen,chefstyle"
+    wrap_ruby_bin "knife"
+
+    appbundle "inspec-bin" "changelog,debug,docs,development"
+    wrap_ruby_bin "inspec"
+
+    appbundle ohai "changelog"
+    wrap_ruby_bin "ohai"
+
+    appbundle "foodcritic" "development,test"
+    wrap_ruby_bin "foodcritic"
+
+    appbundle "test-kitchen" "changelog,debug,docs,development"
+    wrap_ruby_bin "kitchen"
+
+    appbundle "berkshelf" "changelog,debug,docs,development"
+    wrap_ruby_bin "berks"
+
+    appbundle cookstyle "changelog"
+    wrap_ruby_bin "cookstyle"
+
+    appbundle "chef-vault" "changelog"
+    wrap_ruby_bin "chef-vault"
+
+    appbundle "opscode-pushy-client" "changelog"
+    wrap_ruby_bin "pushy-client"
+    wrap_ruby_bin "push-apply"
+    wrap_ruby_bin "pushy-service-manager"
+
+    appbundle chef-apply "changelog,docs,debug" # really, chef-run
+    wrap_ruby_bin "chef-run"
+  )
+
+  if [ "$(readlink /usr/bin/env)" = "$(pkg_interpreter_for core/coreutils bin/env)" ]; then
+    build_line "Removing the symlink created for '/usr/bin/env'"
+    rm /usr/bin/env
+  fi
+
+  build_line "Installing 'chef-analyze' binary"
+  cp "$(pkg_path_for chef-analyze)/bin/chef-analyze" "$pkg_prefix/bin"
+}
+
+appbundle() {
+  build_line "AppBundling gem: '$1' without: '$2'"
+  bundle exec appbundler . "$ruby_bin_dir" "$1" --without "$2" >/dev/null
+}
+
+do_end() {
+  do_default_end
+  # Don't leave this behind - our bundle options will create this
+  # owned by root, making it an annoying cleanup if you're also
+  # doing a 'bundle install' outside of hab.
+  rm -rf "${SRC_PATH}/components/gems/.bundle"
+}
+
+# Stubs
+do_strip() {
+  return 0
+}
+
+# Copied from https://github.com/habitat-sh/core-plans/blob/f84832de42b300a64f1b38c54d659c4f6d303c95/bundler/plan.sh#L32
+wrap_ruby_bin() {
+  local bin_basename
+  local real_cmd
+  local wrapper
+  bin_basename="$1"
+  real_cmd="$ruby_bin_dir/$bin_basename"
+  wrapper="$pkg_prefix/bin/$bin_basename"
+
+  build_line "Adding wrapper for $bin_basename."
```

Comment:
Could we do these on one line to reduce output volume? eg `Adding wrapper for $bin_basename: $wrapper -> $real_cmd`

Reply:
I thought it was better to have a multiline for readability but I could go either way here:

### Multiline:
```
   chef-workstation: Adding wrapper for chef-client.
   chef-workstation:  - from: /hab/pkgs/afiune/chef-workstation/0.10.17/20191008190207/bin/chef-client
   chef-workstation:  -   to: /hab/pkgs/afiune/chef-workstation/0.10.17/20191008190207/ruby-bin/chef-client
```
### Single line:
```
   chef-workstation: Adding wrapper for chef-client: /hab/pkgs/afiune/chef-workstation/0.10.17/20191008190207/bin/chef-client -> /hab/pkgs/afiune/chef-workstation/0.10.17/20191008190207/ruby-bin/chef-client
```